CHANGELOG.md

```diff
@@ -1,10 +1,41 @@
 # HDMF Changelog
 
+## HDMF 4.0.0 (Future)
+
+### New Features
+- Added utility functions and classes useful for implementation of I/O backend to eases maintainability and reuse of functionality across I/O backends. @oruebel (#697)
+  - Added `HDF5IODataChunkIteratorQueue` class in `hdmf.backends.hdf5.h5_utils` as a new helper class for managing the write of `DataChunkIterators` to HDF5 @oruebel (#697)
+  - Added new `hdmf/io/utils.py` module with new I/O utility classes: 1) `WriteStatusTracker` as a simple data structure for tracking the write status of `Builders`, and 2) `NamespaceToBuilderHelper` to help with converting of a namespace to a `Builder` for I/O. @oruebel (#697)
+- Added `get_min_bounds` function to `hdmf.data_utils.DataChunk`. This functionality was originally part of `HDF5IO.__write_chunk__()` and has been moved here to enable reuse of the code across data backends. @oruebel (#697)
+- Added `ignore_string_to_byte` option for `TestCase.assertContainerEqual` to allow us to ignore conversion from string to bytes and just compare unicode values, as different backends may store strings differently. @oruebel (#697)
+
+### Enhancements
+- Updated `HDF5IO` to use the new `WriteStatusTracker`,`NamespaceToBuilderHelper`, and `HDF5IODataChunkIteratorQueue` helper classes. @oruebel (#697)
+  - Removed the following internal helper functions from `HDF5IO` as the functionality is now implemented by the helper classes: `HDF5IO.__convert_namespace`, `HDF5IO.__get_name`, `HDF5IO.__copy_spec`, `HDF5IO.__get_new_specs`, `HDF5IO.__builderhash`, `HDF5IO.__exhaust_dcis`, `HDFIO.__write_chunk__`
```

Comment:
We need not detail changes to private methods. It clutters the changelog which is for public use. These details can live on in the git history.

Reply:
Thats's fine. I was using the changelog in part to keep track of the main changes, but removing this is fine since its just internal details.